enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,574 @@
+---
+title: declarative-index-config
+authors:
+  - "@anik120"
+reviewers:
+  - @njhale
+  - @benluddy
+  - @dmesser
+  - @exdx
+  - @Jamstah
+  - @bparees
+approvers:
+  - TBD
+creation-date: 2020-07-10
+last-updated: 2020-07-10
+status: implementable  
+---
+
+
+# Index representation and management 
+
+### Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+### Summary 
+
+This enhancement proposes the representation of packages in an index in a standard declarative way, so that index authors and operator authors can communicate with each other about the content of an index using json/yaml, without needing to communicate with each other using artifacts(operator manifests etc). Using the declarative representation of the packages in the index, it also proposes allowing index authors and individual package authors to alter the content of the packages(package name, channel information etc) as well as the structure of the packages (channel upgrade graphs etc) of the index without having to rebuild bundles+republish the index.
+
+### Motivation
+
+In the current state of the world, operators are introduced to operator-lifecycle-manager by representing them as [Operator Bundles](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#operator-bundle). These operator bundles are then loaded onto a a container using the [opm](https://github.com/operator-framework/operator-registry/blob/master/docs/design/opm-tooling.md) tool (`opm index add --bundles`) to build an index of operators. Operator bundles conceptually belong to a channel in a package, with each bundle having the ability to upgrade to a different bundle in the channel. 
```

Comment:
Users can integrate with OLM by using SDK. See  https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/Makefile#L108-L119.

- For SDK projects, `make bundle` and `make bundle-build` creates the bundle and the docker image with the indexes. 

- For Non-SDK projects, users are able to use the same command used in these targets to generate the bundle as well based on the manifests. See: `operator-sdk generate bundle -h`